examples/ransomware_detection/stages/preprocessing.py

```diff
@@ -68,44 +72,49 @@ def name(self) -> str:
     def supports_cpp_node(self):
         return False
 
+    def accepted_types(self) -> typing.Tuple:
+        """
+        Returns accepted input types for this stage.
+        """
+        return (AppShieldMessageMeta, )
+
+
     def _sliding_window_offsets(self, ids: typing.List[int], ids_len: int,
-                                window: int) -> typing.List[typing.Tuple[int]]:
+                                window: int) -> typing.List[typing.List[int]]:
         """
         Create snapshot_id's sliding sequence for a given window
         """
-        assert ids_len == len(ids)
-        assert ids_len >= window
 
         sliding_window_offsets = []
 
         for start in range(ids_len - (window - 1)):
             stop = start + window
             sequence = ids[start:stop]
-            consecutive = sorted(sequence) == list(range(min(sequence), max(sequence) + 1))
+            consecutive = sequence == list(range(min(sequence), max(sequence) + 1))
             if consecutive:
                 sliding_window_offsets.append((start, stop))
 
         return sliding_window_offsets
 
-    def _rollover_pending_snapshots(self,
-                                    snapshot_ids: typing.List[int],
-                                    source_pid_process: str,
-                                    snapshot_df: pd.DataFrame):
+    def _rollover_pending_snapshots(self, source_pid_process: str, snapshots_dict):
```

Comment:
Could we also consider adding a time duration to retain the pending rollover snapshots in memory? Without this, they will remain in memory and keep waiting for the sequence (in case if sequences were not ingested to the pipeline)

Reply:
Hi @bsuryadevara ,
I added a function that checks if the scan_ids are consecutively ordered.. like: [1,2,3]...
I think that we don't have pending snapshots in the memory... (I printed them, and they were always cleaned)...

Reply:
I saw in the past, that we can have an anomaly if we have old snapshots. I.e, if we restarted the pipeline but we had snapshots that we don't read from kafka... So it can be problematic...